samples/client/petstore/typescript-fetch/builds/enum/models/StringOneOf.ts

```diff
@@ -0,0 +1,51 @@
+/* tslint:disable */
+/* eslint-disable */
+/**
+ * Enum test
+ * No description provided (generated by Openapi Generator https://github.com/openapitools/openapi-generator)
+ *
+ * The version of the OpenAPI document: 1.0.0
+ * 
+ *
+ * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
+ * https://openapi-generator.tech
+ * Do not edit the class manually.
+ */
+
+import {
+    StringEnum,
+    StringEnumTwo,
+    StringEnumFromJSONTyped,
+    StringEnumToJSON,
+    StringEnumTwoFromJSONTyped,
+    StringEnumTwoToJSON,
+} from './';
+
+/**
+ * @type StringOneOf
+ * 
+ * @export
+ */
+export type StringOneOf = StringEnum | StringEnumTwo;
+
+export function StringOneOfFromJSON(json: any): StringOneOf {
+    return StringOneOfFromJSONTyped(json, false);
+}
+
+export function StringOneOfFromJSONTyped(json: any, ignoreDiscriminator: boolean): StringOneOf {
+    if ((json === undefined) || (json === null)) {
+        return json;
+    }
+    return StringEnumFromJSONTyped(json, true) || StringEnumTwoFromJSONTyped(json, true);
```

Comment:
Previously this came out as a object spread of the wrong type `{...FromJSONTyped(json, true), ...FromJSONTyped(json, true) }`

Reply:
is it valid for an enum to include an empty string (`""`) as a value? If so, we might want to be stricter here and use `??` instead of `||`

Reply:
however, apart from an empty string, this `||` will never work at all. `StringEnumFromJSONTyped` always returns some value (because a null check is done just before it - line 36), so the second part, `StringEnumTwoFromJSONTyped` will never be executed

Reply:
`""` are valid in OAS so `??` is required ... it's also a problem for boolean and numbers so good catch.

I wrongly assumed casting `json as {{classname}}` would return falsy but that's not the case!